src/export_edf.jl

```diff
@@ -118,7 +190,9 @@ function onda_samples_to_edf_signals(onda_samples::AbstractVector{<:Samples}, se
                                                  extrema.physical_min, extrema.physical_max,
                                                  extrema.digital_min, extrema.digital_max,
                                                  "", sample_count)
-            sample_data = vec(samples[channel_name, :].data)
+            # manually convert here in case we have input samples whose encoded
+            # values are convertible losslessly to Int16:
+            sample_data = Int16.(vec(samples[channel_name, :].data))
```

Comment:
this seems unlikely but I guess it doesn't hurt 🤷

Reply:
ah, this is actually necessary when you have input samples with `Int8`/`UInt8` encoding (which is the "mildly cursed" bit you called out above)